### DISCUSSION TITLE
How to open Page on Startup, which will than be recognized by OnNavigatedTo in ViewModel

### DISCUSSION THREAD

**Discussion by anzeko123** [PrismLibrary/Prism#2962] [.NET MAUI] [answered]
Hi,
I am currently trying to open a page when you start the application (yes it opens without this). I need to open it so I can use `OnNavigatedTo` method in a ViewModel so I can initialize my local database.
I am curently migrating from Xamarin.Forms to .NET MAUI. 
In Xamarin.Forms u write the initialization in App.xaml.cs in the `OnInitialized` method.
This is the code in Xamarin.Forms:
```
        protected override async void OnInitialized()
        {
            InitializeComponent();
            await NavigationService.NavigateAsync(new Uri("http://url.com/SignInPage", UriKind.Absolute),
                new NavigationParameters
                {
                    {"OnSuccess",new Uri("http://url.com/SignInPage?IsNew=true", UriKind.Absolute) }
                });
        }
```
I looked through the whole documentation on Navigation in Prism but I couldn't figure out how to make this navigation. In .NET MAUI you don't have the `OnInitialized` method and I think that you can make this navigation in the App.xaml.cs in .NET MAUI.
Is there a way to do this in the same way as you did in Xamarin.Forms.
I tried to create this navigation builder for .NET MAUI but i don't understand it correctly how to set it up and how it works.
I also tried making `INavigationService` as a variable but I get the error for `Object reference not set to an instance of an object` because I don't populate the `NavigationService`
```
public partial class App : Application {
    public App() {
        //public INavigationService NavigationService { get; set; }
        InitializeComponent();
        MainPage = new AppShell();
        NavigationService.NavigateAsync(new Uri("http://url.com/SignInPage", UriKind.Absolute),
            new NavigationParameters
            {
                    {"OnSuccess",new Uri("http://url.com/SignInPage?IsNew=true", UriKind.Absolute) }
            });
    }
}
```
After the page navigates I want to be recognized by OnNavigatedTo.

Thank you to everyone who tries to help
Anzeko

**Comment by dansiegel** [comment, marked as answer]
OnInitialized does still exist though it is the improper place to Navigate in .NET MAUI. You will want to use OnAppStart which is clearly documented [here](https://prismlibrary.com/docs/maui/appbuilder.html#onappstart) and is used the in repo test app.

https://github.com/PrismLibrary/Prism/blob/7c69cfc243d4df195928e46b9146e995ca3ea626/e2e/Maui/PrismMauiDemo/MauiProgram.cs#L53-L55

**Reply by anzeko123** [reply]
Thank you. This works and it properly navigates. Still it does not open the OnNavigatedTo but I thing there is a problem somewhere else.
